hibernate-core/src/main/java/org/hibernate/query/sqm/internal/QuerySqmImpl.java

```diff
@@ -248,11 +250,11 @@ public QuerySqmImpl(
 			SharedSessionContractImplementor producer) {
 		super( producer );
 		this.hql = CRITERIA_HQL_STRING;
-		this.sqm = criteria;
+		this.sqm = criteria.copy( SqmCopyContext.simpleContext() );;
```

Comment:
This should be opted-in to.  As a user, I'd hate to incur this overhead just because the JPA spec says this should be supported.  It is an awful expectation imo